### PR TITLE
fix: extra filters for chart data endpoint

### LINKED ISSUE #10348: Filter isn't applied to word cloud chart on a dashboard
If you apply a filter to a dashboard that has a word cloud chart on it, the word cloud chart will not be filtered.

### Expected results

When a filter is applied to a dashboard, the word cloud chart should have the filter applied as well.

### Actual results

The filter is not applied and the query sent to the backend does not include the filter. However, if you click on the ellipsis menu and "Explore" the word cloud chart in the chart editor, then the filter shows up and is applied.

#### How to reproduce the bug

1. Create a filter box
2. Create a word cloud chart that uses the same datasource as the filter box
3. Create a dashboard and place the filter box and the word cloud chart on it.
4. Apply a filter using the filter box.
5. The filter is not applied to the word cloud chart

### Environment

- superset version: 0.37.0 RC 2
- python version: 3.6.10
- node.js version: 12.14.1
- npm version: 6.13.4

### Checklist

Make sure these boxes are checked before submitting your issue - thank you!

- [X] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X] I have reproduced the issue with at least the latest released version of superset.
- [X] I have checked the issue tracker for the same issue and I haven't found one similar.

### LINKED ISSUE #10349: Applying a filter and then switching tab on a dashboard results in a blank table chart
If you have a dashboard with tabs, a filter box on the first tab and a table chart on a different tab, when you apply a filter then switch tabs, the table chart will be a blank box. This also affects the world map chart

### Expected results

The table chart should render the data as expected.

### Actual results

The table is a blank box. If you force refresh the chart from the ellipsis menu, then it will render. Also, if you resize the browser window, it will render.

#### Screenshots

Screen Shot 2020-07-16 at 5 45 14 PM
Screen Shot 2020-07-16 at 5 45 22 PM

#### How to reproduce the bug

1. Create a filter box chart
2. Create a table chart using the same datasource as the filter box
3. Create a new dashboard
4. Place tabs at the top of the dashboard as in the screen shot, create 2 tabs
5. On the first tab, place the filter box in a row
6. On the second tab, place the table in a row
7. Save the dashboard
8. Bring up the dashboard, on the first tab, select a filter and apply it
9. Switch to the second tab. The table will be a blank box
10. Resize the browser window and the table renders
11. Try also with the world map chart instead of a table chart

### Environment

- superset version: 0.37.0 RC2
- python version: 3.6.10
- node.js version: 12.14.1
- npm version: 6.13.4

### Checklist

Make sure these boxes are checked before submitting your issue - thank you!

- [X] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X] I have reproduced the issue with at least the latest released version of superset.
- [X] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
Currently Filter Box events aren't being properly reflected in the new chart data endpoint requests. This was because `extra_filters` weren't being properly merged with filters from the chart data payload. The bulk of the fix was done in `superset-ui` ( https://github.com/apache-superset/superset-ui/pull/688 ),  and updates `@superset-ui/chart` to the most recent released version.

### TEST PLAN
CI + new tests.

### SCREENSHOTS
Below are screenshots of a Filter Box with changing legacy charts (Name Line, Multiformat Line) and one new chart (Name Cloud, ECharts Multiformat).

#### BEFORE
Previously Filter Box changes (date range, time grain, gender selector) were NOT reflected in the viz' calling the new chart data endpoint (Name Cloud, ECharts Multiformat).
captured (13)
captured (14)

#### AFTER
As can be seen, changes in the Filter Box are reflected properly in all charts.
captured (12)
captured (11)

### ADDITIONAL INFORMATION

- [x] Has associated issue: closes #10348
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/util/charts/getEffectiveExtraFilters.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ import { DataRecordFilters } from '@superset-ui/chart';
 export default function getEffectiveExtraFilters(filters: DataRecordFilters) {
   return Object.entries(filters).map(([column, values]) => ({
     col: column,
-    op: 'in',
+    op: Array.isArray(values) ? 'IN' : '=',
```
Comment (villebro) [inline superset-frontend/src/dashboard/util/charts/getEffectiveExtraFilters.ts:24]: The current logic generates an invalid filter (picked up by TS linter in `superset-ui`).

File: superset-frontend/spec/javascripts/dashboard/util/getEffectiveExtraFilters_spec.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import getEffectiveExtraFilters from 'src/dashboard/util/charts/getEffectiveExtraFilters';
+
+describe('getEffectiveExtraFilters', () => {
+  it('should create IN operator for arrays', () => {
+    const result = getEffectiveExtraFilters({
+      gender: ['girl'],
+      __time_range: ' : 2000-07-17T00:00:00',
```
Comment (graceguo-supercat) [inline superset-frontend/spec/javascripts/dashboard/util/getEffectiveExtraFilters_spec.js:25]: should be `2020` :)

Comment (villebro) [reply superset-frontend/spec/javascripts/dashboard/util/getEffectiveExtraFilters_spec.js:25]: image

File: superset-frontend/src/dashboard/util/charts/getEffectiveExtraFilters.ts [hunk 3]
```diff
@@ -19,9 +19,11 @@
...
-    op: 'in',
-    val: values,
-  }));
+  return Object.entries(filters)
+    .map(([column, values]) => ({
+      col: column,
+      op: Array.isArray(values) ? 'IN' : '=',
+      val: values,
+    }))
+    .filter(filter => filter.val !== null);
```
Comment (graceguo-supercat) [inline superset-frontend/src/dashboard/util/charts/getEffectiveExtraFilters.ts:28]: when user clear out a filter, it's value will be `null`, and we need to send out this to backend.

Comment (villebro) [reply superset-frontend/src/dashboard/util/charts/getEffectiveExtraFilters.ts:28]: @graceguo-supercat when the filter is cleared, the line will be removed, thus triggering a filter event (tested that it works properly on both legacy and new charts). Having an undefined value for a binary or set filter is also invalid according to the type declarations on `superset-ui`, so if this is the case we probably need to make changes there.

**Review by rusackas — APPROVED** [review]
LGTM!

File: superset-frontend/src/dashboard/util/charts/getEffectiveExtraFilters.ts (hunk 3 above)
Comment (graceguo-supercat) [reply superset-frontend/src/dashboard/util/charts/getEffectiveExtraFilters.ts:28]: agree!

**Review by graceguo-supercat — APPROVED** [review]
LGTM!

**Pull request merged**